### ISSUE TITLE
False positives on jsx-no-useless-fragment, need interpolated values to avoid TS errors

### ISSUE THREAD

**Issue by kingdaro** [yannickcr/eslint-plugin-react#2479]
[I found another similar closed issue](https://github.com/yannickcr/eslint-plugin-react/issues/2432), but it reports different sorts of false positives than those I'm encountering.

In our codebase, we have a pattern of doing `return <>{props.children}</>`, since returning `props.children` from function components causes TS errors when children is defined as `React.ReactNode`. We're trying to avoid using `as any` or similar. Example:

```tsx
function LoaderView<T extends ItemStoreObject>(props: LoaderViewProps<T>) {
  useLazyLoad(props.loader);

  const {
    renderError = renderNothing,
    renderLoading = renderNothing,
    loader,
  } = props;

  // The rule warns against all of these returns,
  // but we need them, otherwise we'll get a type error:
  // "Type '{} | null | undefined' is not assignable to type '...'."
  if (loader.data) return <>{props.render(loader.data)}</>;
  if (loader.isLoading) return <>{renderLoading()}</>;
  if (loader.error) return <>{renderError(loader.error)}</>;

  return null;
}
```
It would be great if the rule could either _not_ report on single interpolated values, or if there were a flag for this case

**Comment by ljharb** [comment]
This sounds like a bug in TS's react types, no?

**Comment by kingdaro** [comment]
It probably is. I guess by the time this issue would have been resolved, a fix might be in TS already, rendering the change useless. The issues to fix the TS bug just seem a little stalled on progress, is all. (https://github.com/DefinitelyTyped/DefinitelyTyped/issues/20356, https://github.com/microsoft/TypeScript/pull/29818)

I'll close. Wrong place to fix this.

**Comment by ljharb** [comment]
In the meantime, an eslint override comment, or a TS ignore comment, might be appropriate.
